lib/if.c

```diff
@@ -264,12 +264,20 @@ void if_update_to_new_vrf(struct interface *ifp, vrf_id_t vrf_id)
 
 		if_dnode = yang_dnode_get(
 			running_config->dnode,
-			"/frr-interface:lib/interface[name='%s'][vrf='%s']/vrf",
+			"/frr-interface:lib/interface[name='%s'][vrf='%s']",
 			ifp->name, old_vrf->name);
 		if (if_dnode) {
-			nb_running_unset_entry(if_dnode->parent);
-			yang_dnode_change_leaf(if_dnode, vrf->name);
-			nb_running_set_entry(if_dnode->parent, ifp);
+			nb_running_unset_entry(if_dnode);
+			lyd_free(if_dnode);
+			running_config->version++;
+		}
+
+		if_dnode = yang_dnode_new_path(
+			running_config->dnode,
+			"/frr-interface:lib/interface[name='%s'][vrf='%s']",
+			ifp->name, vrf->name);
+		if (if_dnode) {
+			nb_running_set_entry(if_dnode, ifp);
```

Comment:
This doesn't solve the problem entirely as the interface child nodes will still have outdated xpaths in the hash table. In theory this could be problematic for isisd since it has an user pointer for `/frr-interface:lib/interface/frr-isisd:isis`. Fixing this would involve doing something similar we do for unsetting user pointers, which consists of iterating over the child nodes recursively. Please look at `nb_running_unset_entry_helper()` for more details.